src/templates/csharp/TestBase.cs

```diff
@@ -443,6 +461,73 @@ public List<AppiumWebElement> FindElementsBy(params By[] locators)
             return FindElementsBy(null, Config.ImplicitWaitInMs, locators);
         }
 
+        /**
+         * Scroll to find best element on scrollable
+         */
+        public AppiumWebElement FindElementOnScrollable(params By[] locators)
+        {
+            var infoJsonString = File.ReadAllText($"../../../test/resources/{GetCurrentCommandId()}.json", Encoding.UTF8);
+            dynamic infoObject = JsonConvert.DeserializeObject(infoJsonString);
+            AppiumWebElement? scrollableElement = null;
+            var swipedToTop = false;
+            var screenSize = GetScreenSize();
+
+            var touchableElement = Utils.Retry<AppiumWebElement>(
+                (attempt) =>
+                {
+                    var foundElement = FindElementBy(locators);
+                    var rect = foundElement.Rect;
+                    if (rect.X < 0 || rect.Y < 0)
+                    {
+                        throw new Exception("Element is found but is not visible");
+                    }
+
+                    return foundElement;
```

Comment:
Xium has the endpoint `/session/:sessionId/element/:elementId/location_in_view` that will automatically scroll to the element to make this visible and then return the element position on the screen. Could you give it a try if needed?

Reply:
The appium script run on both Xium and Appium, so we only use endpoint that both server support